File: screenrecorder/src/main/java/com/sunplus/screenrecorder/media/glutils/ITexture.java
package com.sunplus.screenrecorder.media.glutils;

import android.graphics.Bitmap;
import java.io.IOException;

/**
 * Created by w.feng on 2018/10/11
 * Email: fengweisb@gmail.com
 */
public interface ITexture {
  void release();

  void bind();
  void unbind();

  int getTexTarget();
  int getTexture();

  float[] getTexMatrix();
  void getTexMatrix(float[] matrix, int offset);

  int getTexWidth();
  int getTexHeight();

  void loadTexture(String filePath) throws NullPointerException, IOException;
  void loadTexture(Bitmap bitmap) throws NullPointerException;
}

File: screenrecorder/src/main/java/com/almosting/screenrecorder/media/glutils/ITexture.kt
package com.almosting.screenrecorder.media.glutils

import android.graphics.Bitmap
import java.io.IOException

/**
 * Created by w.feng on 2018/10/11
 * Email: fengweisb@gmail.com
 */
interface ITexture {
  fun release()
  fun bind()
  fun unbind()
  fun getTexTarget(): Int
  fun getTexture(): Int
  fun getTexMatrix(): FloatArray?
  fun getTexMatrix(matrix: FloatArray?, offset: Int)
  fun getTexWidth(): Int
  fun getTexHeight(): Int
  @Throws(NullPointerException::class, IOException::class) open fun loadTexture(
    filePath: String?
  )

  @Throws(NullPointerException::class) fun loadTexture(bitmap: Bitmap?)
}